include/oneapi/dpl/internal/reduce_by_segment_impl.h

```diff
@@ -605,8 +605,8 @@ reduce_by_segment_impl(Policy&& policy, InputIterator1 first1, InputIterator1 la
     auto value_output_buf = keep_value_outputs(result2, result2 + n);
 
     using has_known_identity =
-        typename __dpl_sycl::__has_known_identity<BinaryOperator,
-                                                  typename ::std::iterator_traits<InputIterator2>::value_type>::type;
```

Comment:
during this implementation new issue were found: https://github.com/oneapi-src/oneDPL/issues/1353:
This issue should be fixed before the current PR.

Reply:
The fix has been merged, so I think you can rebase this branch with main.